Remotion/Data/DomainObjects/Infrastructure/DomainObjectTransactionContextStrategy.cs

```diff
@@ -0,0 +1,87 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using Remotion.Data.DomainObjects.DataManagement;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Infrastructure
+{
+  /// <summary>
+  /// Provides actual implementation for the <see cref="DomainObjectTransactionContextExecutor"/>.
+  /// Represents the context of a <see cref="DomainObjects.DomainObject"/> that is associated with a specific <see cref="ClientTransaction"/>.
+  /// </summary>
+  [Serializable]
```

Comment:
Why do we need `[Serializable]`?

Reply:
Great question, I have no clue. Probably something I shouldve talked about during standup -> Basically all of the tests that were going red on TC were due to serializing errors, something wants to serialize the strategy, but I'm not sure what, where and why.

Reply:
Okay got the info now, we need `[Serializable]` as `DomainObject`s can be serialized so every store member needs to be serializable as well.